lib/src/block/block_encoder.rs

```diff
@@ -25,12 +24,12 @@ use super::symbol_statistics::{
 
 pub(crate) fn generate_block_data(
     input: &[u8],
```

Comment:
input is now unused, isn't it?

Reply:
Input is still used to calculate checksum here: https://github.com/torfmaster/ribzip2/pull/17/files#diff-d6d456c99d4991ae954f1fe43616e4f64f165c2ad32e365d265d7e528bd01b3bR31

Checksum is further appended to the output.

Reply:
Ah, I see. It might be a good idea to compute the checksum during RLE, however, I don't know how to do it without creating very convoluted code. If you have an idea, please tell me, otherwise I would leave it for now or compute the checksum outside `generate_block_data` based on the amount of data read during rle. Up to you!